src/Catalyst.Core.Lib.Tests/UnitTests/P2P/IO/Observers/AddFileToDfsRequestObserverTests.cs

```diff
@@ -0,0 +1,194 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Threading;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.FileTransfer;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.IO.Messaging.Correlation;
+using Catalyst.Abstractions.Types;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.IO.Messaging.Correlation;
+using Catalyst.Core.Lib.IO.Messaging.Dto;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Core.Modules.Hashing;
+using Catalyst.Core.Modules.Rpc.Server.IO.Observers;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Wire;
+using Catalyst.Protocol.Rpc.Node;
+using Catalyst.TestUtils;
+using DotNetty.Transport.Channels;
+using FluentAssertions;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using Serilog;
+using TheDotNetLeague.MultiFormats.MultiHash;
+using Xunit;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests.P2P.IO.Observers
+{
+    public sealed class AddFileToDfsRequestObserverTests
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IDisposable' in this class and use the 'Dispose' method to call 'Dispose' on '_manualResetEvent'.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)